src/libcore/ptr.rs

```diff
@@ -10,9 +10,58 @@
 
 // FIXME: talk about offset, copy_memory, copy_nonoverlapping_memory
 
-//! Raw, unsafe pointers, `*const T`, and `*mut T`.
+//! Manually manage memory through raw pointers.
 //!
 //! *[See also the pointer primitive types](../../std/primitive.pointer.html).*
+//!
+//! # Safety
+//!
+//! Many functions in this module take raw pointers as arguments and read from
+//! or write to them. For this to be safe, these pointers must be *valid*.
+//! Whether a pointer is valid depends on the operation it is used for
+//! (read or write), and the extent of the memory that is accessed (i.e.,
+//! how many bytes are read/written). Most functions use `*mut T` and `*const T`
+//! to access only a single value, in which case the documentation omits the size
+//! and implicitly assumes it to be `size_of::<T>()` bytes.
+//!
+//! While we can't yet define whether an arbitrary pointer is valid, there
+//! are a few rules regarding validity:
+//!
+//! * A [null] pointer is *never* valid, not even for accesses of [size zero][zst].
+//! * All pointers (except for the null pointer) are valid for all operations of
+//!   [size zero][zst].
+//! * The result of casting a reference to a pointer is valid for as long as the
+//!   underlying object is live and no reference (just raw pointers) is used to
+//!   access the same memory.
```

Comment:
Is this so? 

Say I have a value that I read from many threads using shared references, but where one thread casts the shared reference into a `*const` and then performs the read via the pointer. Forbidding that would be pretty extreme.

Maybe the situation in which this is UB needs to be clarified here.

Reply:
I tried to say that:

> The guarantees that are provided at this point are very minimal:

> Stronger guarantees will be provided eventually, as the [aliasing] rules are being determined.

How can I be clearer?

Reply:
Maybe just add: "That is, your program might be undefined behavior under these minimal guarantees, but as we add more guarantees its behavior will become defined, so don't rewrite your code just yet and when in doubt, ask on XYZ".